### PR TITLE
[ZEPPELIN-3095] fix UI when paragraphs run sequential

### PR DESCRIPTION
**Pull request by Savalek**
### What is this PR for?
This PR fixes the incorrect behavior of UI during the execution of the RunAllParagraphs. 
Including: allows you to copy the note at runtime (before that, the new note was created only after the end of runAllParagraph), blocks some interface elements.
The following items are blocked:

**Paragraphs Control :**
* Move up
* Move down
* Insert new
* Clone paragraph
* Run all above
* Run all below
* Disable run
* Remove

**Note :**
* Insert new paragraph

### What type of PR is it?
Improvement

### What is the Jira issue?
[Zeppelin-3095](https://issues.apache.org/jira/browse/ZEPPELIN-3095)
[ZP-25]

### GIF
peek 2018-11-12 16-20

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

**Comment by Savalek** [comment]
Travis tests OK - [link](https://travis-ci.org/TinkoffCreditSystems/zeppelin/builds/455014003?utm_source=github_status&utm_medium=notification)

**Comment by Savalek** [comment]
[Travis Tests](https://travis-ci.org/TinkoffCreditSystems/zeppelin/builds/455932788?utm_source=github_status&utm_medium=notification)

**Comment by Leemoonsoo** [comment]
Thanks @Savalek for the contribution. I have a few questions.

1. What happens if a browser opens running notebook? Is it correct that `SEQUENTIAL_RUN_STATUS` message will not be received in this case. Can we handle this case?

2. How is it going to behave when notebook run is triggered by cron scheduler?

**Comment by Savalek** [comment]
@Leemoonsoo, if the user opens the laptop during a sequential launch, everything will work correctly.
But now with the cron scheduler work is incorrect.
I will try to fix it.

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java [hunk 1]
```diff
@@ -783,6 +783,15 @@ public void setInfo(Map<String, Object> info) {
     this.info = info;
   }
 
+  public void setSequentialRunStatus(boolean runStatus) {
+    Map<String, Object> infoMap = getInfo();
+    infoMap.put("isRunningSequentially", runStatus);
```
Comment (zjffdu) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java]: Could we change it to more general word `isRunning`? Because what we want to achieve is disallow any write operation on note when it is running. And we may introduce other kinds of running mode in future.

Comment (Savalek) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java]: No objections

Comment (Savalek) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java]: @zjffdu, done.

File: zeppelin-integration/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java [hunk 2]
```diff
@@ -229,6 +229,8 @@ public void testDisableParagraphRunButton() throws Exception {
           getParagraphStatus(1), CoreMatchers.equalTo("READY")
       );
 
+      driver.navigate().refresh();
+      ZeppelinITUtils.sleep(3000, false);
```
Comment (zjffdu) [inline zeppelin-integration/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java]: Why this change ?

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java [hunk 3]
```diff
@@ -1191,8 +1199,15 @@ private void runAllParagraphs(NotebookSocket conn,
             new TypeToken<List<Map<String, Object>>>() {
             }.getType());
 
-    getNotebookService().runAllParagraphs(noteId, paragraphs, getServiceContext(fromMessage),
-        new WebSocketServiceCallback<Paragraph>(conn));
+    Note note = notebook().getNote(noteId);
+
+    try {
+      note.setRunning(true);
```
Comment (zjffdu) [inline zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: Could you move this operation into `NotebookService` ?

**Comment by zjffdu** [comment]
@Savalek Could you remove `ZP-25` from PR title ?

File: zeppelin-integration/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java (hunk 2 above)
Comment (Savalek) [reply zeppelin-integration/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java]: I added `ng-class="{'disabled':isNoteRunning()}"` in `notebook-actionBar.html` and this test began to fail.
He can not find the button to delete the note. 
I tried many solutions.
Only restarting the page helps.

File: zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java (hunk 3 above)
Comment (Savalek) [reply zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java]: done

**Comment by zjffdu** [comment]
@Savalek Could you resolve the conflicts ?

**Comment by Savalek** [comment]
I am working to fix the problem

**Comment by Savalek** [comment]
@zjffdu, I'm fixed NPE.

**Comment by zjffdu** [comment]
LGTM, merge if more comments

**Comment by zjffdu** [comment]
@Savalek I have merged it into master, could you create another PR for branch-0.8 if you'd like to merge it into branch-0.8 as well ?
